integration_tests/run_pyspark_from_build.sh

```diff
@@ -270,6 +270,11 @@ else
       fi
     fi
 
+    # Set a seed to be used for parallel jobs injecting OOMs in tests, as all tests names must
```

Comment:
It's unclear to me what this has to do with parallel jobs.  This seems to be simply a convenience for picking a new seed each time we run the tests, and it seems like it would work even without this code, as the Python code looking for this env variable defaults to a constant if it can't find the env variable.

Reply:
So when running in parallel, python is asking each of the workers to come up with the list of tests. Each worker then makes up a list of all tests and reports back to the main process, which includes deciding which ones should have the injected OOM. So if workers start with a constant seed that all works but then we are always picking the same injection order, so that's why I wanted to change the seed and ensure all parallel workers use the same one, so that the parallel builds don't fail.

Reply:
If the test names (including "[INJECTED_OOM]") don't match exactly between the workers then the run fails, so that's the main symptom.

Reply:
I'm wondering how the workers are going to get a different ID?  It either picks an environment variable that should be the same across all workers (inherited from the single parent process that kicked off the parallel tests) or if the env doesn't exist, each worker picks the same hardcoded seed.  This will pick a different seed each run, but it's unclear to me how they're going to get a different value without it (they'll get a seed of 1 assuming env is unset or whatever the user set it to).

Reply:
I tweaked things over here: https://github.com/NVIDIA/spark-rapids/pull/7925/commits/8e482c8b31450dd1e44849320216e1082290463a, let me know if that's still unclear